edualert/statistics/tasks.py

```diff
@@ -167,3 +176,246 @@ def get_months_since_academic_calendar_start(current_calendar):
+    # our current index.
+    while query_offset < school_units_count:
+        for school_unit in school_units_qs[query_offset:query_offset + query_batch_size]:
+            with tempfile.NamedTemporaryFile(prefix='edu_report_', suffix='.xslx') as file:
+                # get data for report
+                classes = _compute_report_data_for_school_unit(academic_year, month, school_unit)
+                # write report to temporary file
+                _create_report_xslx(file.name, month_name, classes.values())
+                # send email
+                subject = 'Raport lunar absente - {} {}'.format(month_name, academic_year)
+                content = 'Bună ziua!\n\nAcesta este un raport lunar automat în care veți găsi atașat documentul care' \
+                          ' conține evidența absențelor pentru {} {}.'.format(month_name, academic_year)
+                files = [(file.name, '{}.xlsx'.format(subject))]
+                # render template
+                template = get_template('message.html')
+                template_context = {
+                    'title': subject, 'body': content, 'show_my_account': False, 'signature': 'Echipa EduAlert'
+                }
+                bodies = {
+                    'text/html': template.render(context=template_context)
+                }
+                send_mail_with_attachments(subject, bodies, settings.SERVER_EMAIL, [school_unit.email], files)
+
+        query_offset += query_batch_size
+
+
+def _compute_report_data_for_school_unit(academic_year, month, school_unit):
+    unfounded_absences = Count('absence', filter=Q(absence__is_founded=False, absence__taken_at__year=academic_year, absence__taken_at__month=month))
+    founded_absences = Count('absence', filter=Q(absence__is_founded=True, absence__taken_at__year=academic_year, absence__taken_at__month=month))
+
+    classes_query = StudentCatalogPerSubject.objects \
+        .filter(academic_year=academic_year, study_class__school_unit=school_unit) \
+        .values('subject_name', 'study_class__class_grade', 'study_class__class_letter',
+                'study_class__class_master__full_name') \
+        .annotate(unfounded_absences=unfounded_absences, founded_absences=founded_absences)
+
+    # transform query results
+    classes = {}
+    for row in classes_query:
+        class_name = '{} {}'.format(row['study_class__class_grade'], row['study_class__class_letter'])
+
+        # add class if it doesn't exist
+        if not classes.get(class_name):
+            classes[class_name] = {
+                'class_name': class_name,
+                'class_master': row['study_class__class_master__full_name'],
+            }
+
+        # add subjects if the don't exist
+        clazz = classes[class_name]
+        if not clazz.get('subjects'):
+            clazz['subjects'] = []
+
+        # add the subject to the list
+        clazz['subjects'].append({
```

Comment:
not sure this works properly. A StudyClass has a StudentCatalogPerSubject for each subject and for each student (i.e. n x m catalogs, where n = subjects count, m = students count).